constellations/gemini-admin/src/user/userClass.js

```diff
@@ -0,0 +1,212 @@
+            })
+            .catch((err) =>
+                alert("Could not enroll in class: " + err.response.data)
+            );
+    };
+
+    render = () => {
+        const user = this.state.user;
+
+        var fullName = user.firstName;
+        if (user.middleName) {
+            fullName += " " + user.middleName + " " + user.lastName;
+        } else {
+            fullName += " " + user.lastName;
+        }
+
+        const rows = this.state.userClasses.map((c, index) => {
+            const enrollStatus = this.state.enrollStatuses[index];
+            const fullState = c.fullState;
+            let state = "";
+            let firstSpace = <button></button>;
+            let secondSpace = <button></button>;
+            switch (fullState) {
+                case 0:
+                    state = "Empty";
+                    firstSpace = <button>Reject</button>;
+                    secondSpace = <button>Accept</button>;
+                    break;
+                case 1:
+                    status = "Full";
+                    firstSpace = <button>Reject</button>;
+                    break;
+                case 2:
+                    state = "Almost Full";
+                    firstSpace = <button>Reject</button>;
+                    secondSpace = <button>Accept</button>;
+                    break;
+            }
+            if (enrollStatus == "Enrolled") {
+                firstSpace = <button></button>;
+                secondSpace = <button>Unenroll</button>;
+            }
+
+            return (
+                <div className="row" key={index}>
+                    <span className="large-column">{c.classId}</span>
+                    <span className="column">{enrollStatus}</span>
+                    <span className="column">{state}</span>
+                    <span className="button">{firstSpace}</span>
+                    <span className="button">{secondSpace}</span>
+                </div>
+            );
+        });
+
+        const classOptions = this.state.classes.map((c, index) => {
+            return <option key={index}>{c.classId}</option>;
+        });
+
+        var enroll = (
+            <span>
+                There are no classes to choose from. Please add one{" "}
+                <Link to="/classes/add">here</Link>
+            </span>
+        );
+        if (this.state.classes.length != 0) {
+            enroll = (
+                <div>
+                    <p>Select a Class ID to enroll user into:</p>
+                    <select
+                        value={this.state.classId}
+                        onChange={(e) => this.onClassChange(e)}>
+                        <option default hidden>
+                            Select a classId
+                        </option>
+                        {classOptions}
+
+                        <button onClick={this.onClickEnroll}>Enroll</button>
+                    </select>
+                </div>
+            );
+        }
```

Comment:
Maybe also consider a small component or function that extracts this away from the return function.

Then the render function can be more readable if we know it's made up of parts / sections.
```
render = () => {
    const rows = ...;
    const options = ...;
    const enroll = renderEnrollSection();
    return (...);
}
```